components/automate-cli/pkg/verifyserver/services/systemresourceservice/systemresourcesservice.go

```diff
@@ -0,0 +1,375 @@
+	var resp *models.Checks
+	if err != nil {
+		srs.logger.Error("Unable to determine free space of /hab :", err)
+		resp = srs.GetChecksModel(false, fmt.Sprintf(FREE_SPACE_CHECK, "Hab"), "", err.Error(), "Please run system on supported platform")
+		return resp, nil
+	}
+
+	srs.logger.Debugf("The current total free space in hab : %0.2fGB", currentFreeSpaceInGB)
+
+	switch nodeType {
+	case constants.AUTOMATE:
+		passed, successMsg, errorMsg, resolutionMsg := srs.GetCheckModelValuesDetailsForStorageChecks(constants.HAB_FREE_DISK_BEFORE_DEP_A2, currentFreeSpaceInGB, 0, constants.HAB_FREE_DISK_BEFORE_DEP_A2, constants.PRE_DEPLOY, "/hab")
+		resp = srs.GetChecksModel(passed, fmt.Sprintf(FREE_SPACE_CHECK, "Hab"), successMsg, errorMsg, resolutionMsg)
+	case constants.CHEF_INFRA_SERVER:
+		passed, successMsg, errorMsg, resolutionMsg := srs.GetCheckModelValuesDetailsForStorageChecks(constants.HAB_FREE_DISK_BEFORE_DEP_CS, currentFreeSpaceInGB, 0, constants.HAB_FREE_DISK_BEFORE_DEP_CS, constants.PRE_DEPLOY, "/hab")
+		resp = srs.GetChecksModel(passed, fmt.Sprintf(FREE_SPACE_CHECK, "Hab"), successMsg, errorMsg, resolutionMsg)
+	case constants.OPENSEARCH:
+		passed, successMsg, errorMsg, resolutionMsg := srs.GetCheckModelValuesDetailsForStorageChecks(constants.HAB_FREE_DISK_BEFORE_DEP_OS, currentFreeSpaceInGB, 0, constants.HAB_FREE_DISK_BEFORE_DEP_OS, constants.PRE_DEPLOY, "/hab")
+		resp = srs.GetChecksModel(passed, fmt.Sprintf(FREE_SPACE_CHECK, "Hab"), successMsg, errorMsg, resolutionMsg)
+	case constants.POSTGRESQL:
+		passed, successMsg, errorMsg, resolutionMsg := srs.GetCheckModelValuesDetailsForStorageChecks(constants.HAB_FREE_DISK_BEFORE_DEP_PG, currentFreeSpaceInGB, 0, constants.HAB_FREE_DISK_BEFORE_DEP_PG, constants.PRE_DEPLOY, "/hab")
+		resp = srs.GetChecksModel(passed, fmt.Sprintf(FREE_SPACE_CHECK, "Hab"), successMsg, errorMsg, resolutionMsg)
+	case constants.BASTION:
+		passed, successMsg, errorMsg, resolutionMsg := srs.GetCheckModelValuesDetailsForStorageChecks(constants.HAB_FREE_DISK_BEFORE_DEP_BASTION, currentFreeSpaceInGB, 0, constants.HAB_FREE_DISK_BEFORE_DEP_BASTION, constants.PRE_DEPLOY, "/hab")
+		resp = srs.GetChecksModel(passed, fmt.Sprintf(FREE_SPACE_CHECK, "Hab"), successMsg, errorMsg, resolutionMsg)
+	default:
+		srs.logger.Errorf(INVALID_NODE_TYPE_ERR, nodeType)
+		return nil, fmt.Errorf(INVALID_NODE_TYPE_ERR, nodeType)
+	}
+	return resp, nil
+}
+
+func (srs *SystemResourcesService) GetHabFreeSpaceCheckPostDeployment(nodeType string) (*models.Checks, error) {
+	srs.logger.Debugf("Hab free space check is running for node_type : %s  and deployment_state : %s", nodeType, constants.POST_DEPLOY)
+	currentFreeSpaceInGB, err := srs.GetFreeDiskSpaceOfGivenDir("/hab")
+
+	var resp *models.Checks
+	if err != nil {
+		srs.logger.Error("Unable to determine free space of /hab :", err.Error())
+		resp = srs.GetChecksModel(false, fmt.Sprintf(FREE_SPACE_CHECK, "Hab"), "", err.Error(), "Please run system on supported platform")
```

Comment:
failed cases are not covered in unit tests pls add them

Reply:
Done for possible lines